File: src/main/java/net/thedragonteam/thedragonlib/util/TextHelper.java
package net.thedragonteam.thedragonlib.util;
//General place to call text formatting from. You can alternatively use 'EnumChatFormatting'.


import net.minecraft.client.resources.I18n;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import org.apache.commons.lang3.text.WordUtils;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.IntStream;

@SideOnly(Side.CLIENT)
public class TextHelper {
    public static String getFormattedText(String string) {
        return string.replaceAll("&", "\u00A7");
    }

    public static String localize(String input, Object... format) {
        return I18n.format(input, format);
    }

    public static String localizeEffect(String input, Object... format) {
        return getFormattedText(localize(input, format));
    }

    public static String[] localizeAll(String[] input) {
        return Arrays.stream(input).map(s -> localize(s)).toArray(String[]::new);
    }

    public static String[] localizeAllEffect(String[] input) {
        return Arrays.stream(input).map(s -> localizeEffect(s)).toArray(String[]::new);
    }

    public static ArrayList<String> localizeAll(List<String> input) {
        ArrayList<String> ret = new ArrayList<>(input.size());
        IntStream.range(0, input.size()).forEachOrdered(i -> ret.add(i, localize(input.get(i))));
        return ret;
    }

    public static ArrayList<String> localizeAllEffect(List<String> input) {
        ArrayList<String> ret = new ArrayList<>(input.size());
        IntStream.range(0, input.size()).forEachOrdered(i -> ret.add(i, localizeEffect(input.get(i))));
        return ret;
    }

    public static String[] cutLongString(String string, int characters) {
        return WordUtils.wrap(string, characters, "/cut", false).split("/cut");
    }

    public static String[] cutLongString(String string) {
        return cutLongString(string, 30);
    }
}

File: src/main/java/net/thedragonteam/thedragonlib/util/TextHelper.kt
package net.thedragonteam.thedragonlib.util

//General place to call text formatting from. You can alternatively use 'EnumChatFormatting'.


import net.minecraft.client.resources.I18n
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import org.apache.commons.lang3.text.WordUtils
import java.util.*

@SideOnly(Side.CLIENT)
object TextHelper {
    fun getFormattedText(string: String): String {
        return string.replace("&".toRegex(), "\u00A7")
    }

    fun localize(input: String, vararg format: Any): String {
        return I18n.format(input, *format)
    }

    fun localizeEffect(input: String, vararg format: Any): String {
        return getFormattedText(localize(input, *format))
    }

    fun localizeAll(input: Array<String>): Array<String?> {
        val ret = arrayOfNulls<String>(input.size)
        for (i in input.indices)
            ret[i] = localize(input[i])

        return ret
    }

    fun localizeAllEffect(input: Array<String>): Array<String?> {
        val ret = arrayOfNulls<String>(input.size)
        input.indices.forEach { i -> ret[i] = localizeEffect(input[i]) }

        return ret
    }

    fun localizeAll(input: List<String>): ArrayList<String> {
        val ret = ArrayList<String>(input.size)
        input.indices.forEach { i -> ret.add(i, localize(input[i])) }

        return ret
    }

    fun localizeAllEffect(input: List<String>): ArrayList<String> {
        val ret = ArrayList<String>(input.size)
        input.indices.forEach { i -> ret.add(i, localizeEffect(input[i])) }

        return ret
    }

    @JvmOverloads fun cutLongString(string: String, characters: Int = 30): Array<String> {
        return WordUtils.wrap(string, characters, "/cut", false).split("/cut".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
    }
}